README.md

```diff
@@ -605,13 +606,15 @@ Then, you will need to use the `e` option to manually choose which lines to add.
 <a href="unstaging-edits-and-staging-the-unstaged"></a>
 ### I want to stage my unstaged edits, and unstage my staged edits
 
-This is tricky. The best I figure is that you should stash your unstaged edits. Then, reset. After that, pop your stashed edits back, and add them.
+In many cases, You should unstage all files and then pick file you want and commit it.
+But assume that's what you want,You can create a temp commit to store your stage files, stage your unstaged files and stash them, Then, reset the temp commit and stash pop.
 
 ```sh
-$ git stash -k
-$ git reset --hard
-$ git stash pop
-$ git add -A
+$ git commit -m "WIP"
+$ git add . # also have untracked files
+$ git stash
+$ git reset HEAD^
```

Comment:
Why Head^ over --hard?

Reply:
`--hard` is involved HEAD, index and files. I don't want to lost my files.Only rollback the last commit marked "WIP" to get my staged files.

Reply:
Ah, I see what you're doing now. That makes sense, actually, and is easier. Maybe we could keep both ways in this document?

Reply:
I could, but the old way is wrong.You will mark everything staged.